### PR TITLE
feat(remix): Support `v2_errorBoundary` future flag

### PR DESCRIPTION
**Pull request by anagstef**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [x] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

This PR introduces `V2_ClerkErrorBoundary` to support `v2_errorBoundary` future flag

Users that have enabled the `v2_errorBoundary` future flag need to use `V2_ClerkErrorBoundary` in their `root.tsx` page:

```ts
export const ErrorBoundary = V2_ClerkErrorBoundary();
```

### CODE REVIEW

**Comment by anagstef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @anagstef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.23.7 |
| @clerk/chrome-extension | 0.3.20-snapshot.6b46a7e |
| @clerk/clerk-js | 4.51.1-snapshot.6b46a7e |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.11-snapshot.6b46a7e |
| @clerk/fastify | 0.5.6 |
| gatsby-plugin-clerk | 4.3.19 |
| @clerk/localizations | 1.22.0 |
| @clerk/nextjs | 4.21.13 |
| @clerk/clerk-react | 4.21.1 |
| @clerk/remix | 2.6.16 |
| @clerk/clerk-sdk-node | 4.10.14 |
| @clerk/shared | 0.19.1 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.46.1 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.23.7
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.3.20-snapshot.6b46a7e
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.51.1-snapshot.6b46a7e
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.11-snapshot.6b46a7e
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.5.6
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.3.19
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.22.0
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.21.13
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.21.1
```
```sh
# @clerk/remix
npm i @clerk/remix@2.6.16
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.10.14
```
```sh
# @clerk/shared
npm i @clerk/shared@0.19.1
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.46.1
```

**Comment by anagstef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @anagstef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.23.7 |
| @clerk/chrome-extension | 0.3.20-snapshot.356787e |
| @clerk/clerk-js | 4.51.1-snapshot.356787e |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.11-snapshot.356787e |
| @clerk/fastify | 0.5.6 |
| gatsby-plugin-clerk | 4.3.19 |
| @clerk/localizations | 1.22.0 |
| @clerk/nextjs | 4.21.13 |
| @clerk/clerk-react | 4.21.1 |
| @clerk/remix | 2.7.0-snapshot.356787e |
| @clerk/clerk-sdk-node | 4.10.14 |
| @clerk/shared | 0.19.1 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.46.1 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.23.7
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.3.20-snapshot.356787e
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.51.1-snapshot.356787e
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.11-snapshot.356787e
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.5.6
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.3.19
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.22.0
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.21.13
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.21.1
```
```sh
# @clerk/remix
npm i @clerk/remix@2.7.0-snapshot.356787e
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.10.14
```
```sh
# @clerk/shared
npm i @clerk/shared@0.19.1
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.46.1
```

File: packages/remix/src/client/V2_ClerkErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { isRouteErrorResponse, useRouteError } from '@remix-run/react';
+import React from 'react';
+
+import { Interstitial } from './Interstitial';
+
+/*
+ * V2_ClerkErrorBoundary needs the `v2_errorBoundary` future flag to be enabled.
```
Comment (nikosdouvlis) [inline packages/remix/src/client/V2_ClerkErrorBoundary.tsx:7]:
```suggestion
/**
 * Support for Remix ErrorBoundary v2.
 * You need to have the v2_errorBoundary flag enabled in your Remix config,
 * as shown here: https://remix.run/docs/en/main/route/error-boundary-v2
 *
 * @experimental This API is experimental and might change. This API will become stable
 * once Remix officially releases v2.
 */
```

**Review by nikosdouvlis — APPROVED** [review]

File: packages/remix/src/client/V2_ClerkErrorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ */
+export function V2_ClerkErrorBoundary(RootErrorBoundary?: React.ComponentType) {
```
[the remark this answers is not in the document]
Comment (anagstef) [reply packages/remix/src/client/V2_ClerkErrorBoundary.tsx:9]: In V2 the `CatchBoundary` is removed and all the errors end up in the `ErrorBoundary`. A helper with the name `isRouteErrorResponse` helps you identify if the error was an expected error (`4xx`) or an unexpected error.

File: packages/remix/src/client/index.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 export * from './RemixClerkProvider';
 export { ClerkApp } from './ClerkApp';
 export * from './ClerkCatchBoundary';
+export * from './V2_ClerkErrorBoundary';
```
Comment (nikosdouvlis) [inline packages/remix/src/client/index.ts:4]: Let's explicitly export the API, we need to try and avoid `*` :)

**Review by nikosdouvlis — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**